addon-api/background/MessagePasser.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Handles message passing. Mimics chrome.runtime message passing.
```

Comment:
> Mimics chrome.runtime message passing.

pls don't

`chrome.runtime`'s is too confusing

just do two functions: `addon.messaging.send` and `addon.messaging.onRecieve`. Make them both available in all environments (popup, userscript, and bg)

```js
export default async function({ addon }) {
  addon.messaging.send({
    data: "hi",
    otherData: "hello",
  })
}
```

```js
export default async function({ addon }) {
  addon.messaging.onRecive((data) => console.log(data)) // { data: "hi", otherData: "hello" }
}

```

Reply:
best if `onRecieve` is an event listener, so you can remove it and stuff.
```js
addon.messaging.addEventListener("data", ({ data } => console.log(data)))
```

Reply:
> best if `onRecieve` is an event listener, so you can remove it and stuff.
> 
> ```js
> addon.messaging.addEventListener("data", ({ data } => console.log(data)))
> ```

Okay, but I need to add the `sendResponse` function as well. How would I fix that?

and @apple502j, the same problem if I use `EventTarget`.

Reply:
> Okay, but I need to add the sendResponse function as well. How would I fix that?

By just adding it?

> and @apple502j, the same problem if I use EventTarget.

What's the problem with an event target? All the other addon.* are event targets.

Reply:
Nevermind...

Reply:
Finished.